File: avalon-modules/avalon-authorization/src/main/java/io/github/lishangbu/avalon/authorization/controller/RoleController.java
package io.github.lishangbu.avalon.authorization.controller;

import io.github.lishangbu.avalon.authorization.entity.Role;
import io.github.lishangbu.avalon.authorization.service.RoleService;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/// 角色控制器
///
/// 提供角色相关的 REST 接口（占位）
///
/// @author lishangbu
/// @since 2025/8/30
@RequestMapping("/role")
@RestController
@RequiredArgsConstructor
public class RoleController {

    private final RoleService roleService;

    /// 分页条件查询角色
    ///
    /// @param pageable 分页参数
    /// @param role     查询条件
    /// @return 角色分页结果
    @GetMapping("/page")
    public Page<Role> getRolePage(Pageable pageable, Role role) {
        return roleService.getPageByCondition(role, pageable);
    }

    /// 条件查询角色列表
    ///
    /// @param role 查询条件
    /// @return 角色列表
    @GetMapping("/list")
    public List<Role> listRoles(Role role) {
        return roleService.listByCondition(role);
    }

    /// 根据 ID 查询角色
    ///
    /// @param id 角色 ID
    /// @return 角色信息
    @GetMapping("/{id:\\d+}")
    public Role getById(@PathVariable Long id) {
        return roleService.getById(id).orElse(null);
    }

    /// 新增角色
    ///
    /// @param role 角色实体
    /// @return 保存后的角色
    @PostMapping
    public Role save(@RequestBody Role role) {
        return roleService.save(role);
    }

    /// 更新角色
    ///
    /// @param role 角色实体
    /// @return 更新后的角色
    @PutMapping
    public Role update(@RequestBody Role role) {
        return roleService.update(role);
    }

    /// 根据 ID 删除角色
    ///
    /// @param id 角色 ID
    @DeleteMapping("/{id:\\d+}")
    public void deleteById(@PathVariable Long id) {
        roleService.removeById(id);
    }
}

File: avalon-modules/avalon-authorization/src/main/kotlin/io/github/lishangbu/avalon/authorization/controller/RoleController.kt
package io.github.lishangbu.avalon.authorization.controller

import io.github.lishangbu.avalon.authorization.entity.dto.RoleSpecification
import io.github.lishangbu.avalon.authorization.entity.dto.RoleView
import io.github.lishangbu.avalon.authorization.entity.dto.SaveRoleInput
import io.github.lishangbu.avalon.authorization.entity.dto.UpdateRoleInput
import io.github.lishangbu.avalon.authorization.service.RoleService
import jakarta.validation.Valid
import org.babyfish.jimmer.Page
import org.springframework.data.domain.Pageable
import org.springframework.web.bind.annotation.*

/**
 * 角色控制器
 *
 * 提供角色管理相关接口
 *
 * @author lishangbu
 * @since 2025/8/30
 */
@RequestMapping("/role")
@RestController
class RoleController(
    /** 角色服务 */
    private val roleService: RoleService,
) {
    /**
     * 分页条件查询角色
     *
     * @param pageable 分页参数
     * @param role 查询条件
     * @return 角色分页结果
     */
    @GetMapping("/page")
    fun getRolePage(
        pageable: Pageable,
        @ModelAttribute specification: RoleSpecification,
    ): Page<RoleView> = roleService.getPageByCondition(specification, pageable)

    /**
     * 条件查询角色列表
     *
     * @param role 查询条件
     * @return 角色列表
     */
    @GetMapping("/list")
    fun listRoles(
        @ModelAttribute specification: RoleSpecification,
    ): List<RoleView> = roleService.listByCondition(specification)

    /**
     * 根据 ID 查询角色
     *
     * @param id 角色 ID
     * @return 角色信息
     */
    @GetMapping("/{id:\\d+}")
    fun getById(
        @PathVariable id: Long,
    ): RoleView? = roleService.getById(id)

    /**
     * 新增角色
     *
     * @param input 角色写入请求
     * @return 保存后的角色
     */
    @PostMapping
    fun save(
        @RequestBody @Valid input: SaveRoleInput,
    ): RoleView = roleService.save(input)

    /**
     * 更新角色
     *
     * @param input 角色写入请求
     * @return 更新后的角色
     */
    @PutMapping
    fun update(
        @RequestBody @Valid input: UpdateRoleInput,
    ): RoleView = roleService.update(input)

    /**
     * 根据 ID 删除角色
     *
     * @param id 角色 ID
     */
    @DeleteMapping("/{id:\\d+}")
    fun deleteById(
        @PathVariable id: Long,
    ) {
        roleService.removeById(id)
    }
}
